ramips: add support for Cudy WR1300 v3

zekica opened the pull request:
Specifications:
 - SoC: MediaTek MT7621AT
 - RAM: 128 MB (DDR3)
 - Flash: 16 MB (SPI NOR)
 - WiFi: MediaTek MT7603E, MediaTek MT7613BE
 - Switch: 1 WAN, 4 LAN (Gigabit)
 - Buttons: Reset, WPS
 - LEDs: System, Wan, Lan 1-4, WiFi 2.4G, WiFi 5G, WPS
 - Power: DC 12V 1A tip positive

Download and flash the manufacturer's built OpenWRT image available at http://www.cudytech.com/openwrt_software_download
Install the new OpenWRT image via luci (System -> Backup/Flash firmware) Be sure to NOT keep settings. The force upgrade may need to be checked due to differences in router naming conventions.

Cudy WR1300 v3 differs from v2 only in swapped WiFi chip PCIe slots.

The same manufacturer's built OpenWRT image is provided for both v2 and v3 devices as a step in installing, but for proper WiFi functionality, a separate build is required.

Recovery:
 - Loads only signed manufacture firmware due to bootloader RSA verification
 - serve tftp-recovery image as /recovery.bin on 192.168.1.88/24
 - connect to any lan ethernet port
 - power on the device while holding the reset button
 - wait at least 8 seconds before releasing reset button for image to download
 - See http://www.cudytech.com/newsinfo/547425.html

zekica commented:
@DragonBluep as discussed, this is a pull request for supporting the v3 variant.

neheb commented:
If that's the only difference, would make sense to move stuff into a dtsi.

zekica commented:
@neheb @DragonBluep 

I have extracted common nodes to a single .dtsi and updated v2 and v3 dts files to only contain nodes that differ.

The new build boots and works fine on a v3 device, and when decompiling .dtb for v2 it only differs in eeprom_factory_8000 reg length (from the fix in the previous PR) and I have added ieee80211-freq-limit.

DragonBluep commented on target/linux/ramips/dts/mt7621_cudy_wr1300-v2v3.dtsi:
```diff
@@ -0,0 +1,182 @@
+				read-only;
+			};
+
+			factory: partition@40000 {
```
`factory` label can be removed.

DragonBluep commented on target/linux/ramips/dts/mt7621_cudy_wr1300-v2v3.dtsi:
```diff
@@ -0,0 +1,182 @@
+				read-only;
+			};
+
+			bdinfo: partition@ff0000 {
```
`bdinfo` label can be removed.

DragonBluep reviewed (commented):
Please `git rebase` to master branch to squash all commits into one. 'merge' commit is unacceptable.

DragonBluep commented:
By the way, you don't need to open a new PR, just `git push --force`.

zekica commented:
> By the way, you don't need to open a new PR, just `git push --force`.

I did the following:
- stashed my changes
- reverted to before the initial commit
- updated to the latest commit from upstream
- created a new commit on top of that
- force pushed

But somehow github automatically closed the pull request.
